src/gradescopeapi/api/api.py

```diff
@@ -1,200 +1,359 @@
 
     Returns:
-        TokenModel: _description_
+        connection (GSConnection): an instance of the GSConnection class,
+            containing the session object used to make HTTP requests,
+            a boolean defining True/False if the user is logged in, and
+            the user's Account object.
     """
-    pass
-
+    return connection
 
-@app.get("/account")
-def get_courses(token: TokenModel):
-    """_summary_
 
-    Args:
-        token (TokenModel): _description_
+def get_gs_connection_session():
+    """
+    Returns session of the the GSConnection instance
 
     Returns:
-        CourseModel: _description_
+        connection.session (GSConnection.session): an instance of the GSConnection class' session object used to make HTTP requests
     """
-    pass
-
+    return connection.session
 
-@app.get("/courses/{course_id}/memberships")
-def get_course_memberships(token: TokenModel, course_id: str):
-    """_summary_
 
-    Args:
-        token (TokenModel): _description_
-        course_id (str): _description_
+def get_account():
+    """
+    Returns the user's Account object
 
     Returns:
-        UserModel: _description_
+        Account (Account): an instance of the Account class, containing
+            methods for interacting with the user's courses and assignments.
     """
-    pass
-
+    return Account(session=get_gs_connection_session)
 
-@app.get("/courses/{course_id}")
-def get_assignments(token: TokenModel, course_id: str):
-    """_summary_
-
-    Args:
-        token (TokenModel): _description_
-        course_id (str): _description_
 
-    Returns:
-        AssignmentModel: _description_
-    """
-    pass
+@app.get("/")
+def root():
+    return {"message": "Hello World"}
 
 
-@app.get("/courses/{course_id}/assignments/{assignment_id}/submissions/{submission_id}")
-def get_assignment_submission(
-    token: TokenModel, course_id: str, assignment_id: str, submission_id: str
+@app.post("/login", name="login")
+def login(
+    login_data: LoginRequestModel,
+    gs_connection: GSConnection = Depends(get_gs_connection),
 ):
-    """_summary_
+    """Login to Gradescope, with correct credentials
 
     Args:
-        token (TokenModel): _description_
-        assignment_id (str): _description_
+        username (str): email address of user attempting to log in
+        password (str): password of user attempting to log in
 
-    Returns:
-        SessionModel: _description_
+    Raises:
+        HTTPException: If the request to login fails, with a 404 Unauthorized Error status code and the error message "Account not found".
     """
-    pass
+    user_email = os.getenv("USER_EMAIL")
+    password = os.getenv("PASSWORD")
 
+    try:
+        connection.login(user_email, password)
+        return {"message": "Login successful", "status_code": status.HTTP_200_OK}
+    except ValueError as e:
+        raise HTTPException(status_code=404, detail=f"Account not found. Error {e}")
```

Comment:
Refactor to support multiple accounts

Reply:
will do, adding this support now

also to answer your question about the auto-generated docs: 
in the `api` folder of the project, run `uvicorn api:app --reload` and then navigate to `localhost:8000/docs` to find the docs that FastAPI autogenerates